### PR TITLE
Add stripIgnoredCharacters utility function

### LINKED ISSUE #1523: Feature request: strip whitespace from GraphQL queries / fragments / mutations / subscriptions
Currently, when issuing requests to a GraphQL API, the GraphQL string describing the query contains a lot of whitespace. While this makes readability in development mode easier, it is wasteful in production.

Would it be possible to add a function to this library that strips this whitespace away? Here's some example code that does this:

```js
let graphQLQuery = `...` // some GraphQL literal

graphQLQuery = graphQLQuery
    .replace(/#.*\n/g, '')
    .replace(/[\s|,]*\n+[\s|,]*/g, ' ')
    .replace(/:\s/g, ':')
    .replace(/,\s/g, ',')
    .replace(/\)\s\{/g, '){')
    .replace(/\}\s/g, '}')
    .replace(/\{\s/g, '{')
    .replace(/\s\}/g, '}')
    .replace(/\s\{/g, '{')
    .replace(/\)\s/g, ')')
    .replace(/\(\s/g, '(')
    .replace(/\s\)/g, ')')
    .replace(/\s\(/g, '(')
    .replace(/=\s/g, '=')
    .replace(/\s=/g, '=')
    .replace(/@\s/g, '@')
    .replace(/\s@/g, '@')
    .replace(/\s\$/g, '$')
    .replace(/\s\./g, '.')
    .trim()
```

This code turns a GraphQL query such as this one:

```gql
query SomeQuery($foo: String!, $bar: String) {
  someField(foo: $foo, bar: $bar) {
    a
    b {
      c
      d
    }
  }
} 
```

into:

```gql
query SomeQuery($foo:String!$bar:String){someField(foo:$foo bar:$bar){a b{c d}}}
```

Preferably this would happen at compile time for a production build, so the GraphQL literals in the source code would be minified in the output bundle and thus sent in minified form to the GraphQL API.

Proposed name for this function: `condense`.

### PR DESCRIPTION
**Pull request by IvanGoncharov**
Heavily based on work done by @rybon in #1628.
Fixes #1523.

### CODE REVIEW

File: src/utilities/__tests__/stripIgnoredCharacters-test.js [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+    for (const ignored of ignoredTokens) {
+      expectStripped(ignored).toEqual('');
+
+      for (const anotherIgnored of ignoredTokens) {
+        expectStripped(ignored + anotherIgnored).toEqual('');
+      }
+    }
+
+    expectStripped(ignoredTokens.join('')).toEqual('');
```
Comment (IvanGoncharov) [inline src/utilities/__tests__/stripIgnoredCharacters-test.js]: I wrote tests as a combination of handwritten test that covers algorithm and all branches and fuzzing tests that test all possible combinations of tokens.
Fuzzing tests helped to uncover a few edge cases that I fixed and include in handwritten tests but I think fuzzing tests are still useful in case if we change the code or extend GraphQL language grammar.

File: src/utilities/__tests__/stripIgnoredCharacters-test.js [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+    expectStrippedString('"""\\"""\n"""').toEqual('"""\\""""""');
+
+    expectStrippedString('"""\na\n b"""').toStayTheSame();
+    expectStrippedString('"""\n a\n b"""').toEqual('"""a\nb"""');
+    expectStrippedString('"""\na\n b\nc"""').toEqual('"""a\n b\nc"""');
+
+    // Testing with length >5 is taking exponentially more time however it
+    // highly recommended to test with increased limit if you make any change
+    const maxCombinationLenght = 5;
+    const possibleChars = ['\n', ' ', '"', 'a', '\\'];
```
Comment (IvanGoncharov) [inline src/utilities/__tests__/stripIgnoredCharacters-test.js]: This test helped me to uncovered that
```
"""
a
 a
"""
```
can't be converted into:
```
"""a
 a"""
```
Since you will change string value from `a\n a` into `a\na`.
That's why I think it's important to keep it even if can't run it with length `>5` by default.

**Comment by IvanGoncharov** [comment]
@rybon Sorry for the delay, it took me significantly more time than I expected.
Since some developers want to use this function as part of the hashing algorithm for functions I wanted this function to be well tested and handle all edge cases.
Can you please review this implementation to see if it meets your requirements?

@Cito @martijnwalraven @mjmahone If you have time, can you please review this PR?

**Comment by rybon** [comment]
@IvanGoncharov yes, I will review it now. Thanks again!

**Comment by rybon** [comment]
Merge?

**Comment by IvanGoncharov** [comment]
@rybon Merged 🎉 
I will work on preparing a release next week.

**Comment by rybon** [comment]
@IvanGoncharov could you do a release to NPM?

**Comment by IvanGoncharov** [comment]
@rybon Just returned from GraphQL Asia.
And new release is at the top of the list, I will try to release tomorrow.

**Comment by IvanGoncharov** [comment]
@rybon Sorry for the delay. 
I just published `14.3.0` 📦 that includes this PR 🎉
